packages/wrangler/src/__tests__/dev.test.tsx

```diff
@@ -216,6 +216,28 @@ describe("wrangler dev", () => {
 				})
 			);
 		});
+		it("should use custom domain", async () => {
+			fs.writeFileSync("index.js", `export default {};`);
+
+			// config.routes
+			mockGetZones("customdomain.some-host.com", [{ id: "some-zone-id-5" }]);
+			writeWranglerToml({
+				main: "index.js",
+				routes: [
+					{ pattern: "customdomain.some-host.com", custom_domain: true },
+				],
+			});
+			await runWrangler("dev");
+			expect((Dev as jest.Mock).mock.calls[0][0]).toEqual(
```

Comment:
Since the `createPreviewSession()` function, which is the focus of this fix, is actually only called within the `Dev` component, this test does not check the changes to that function, since `Dev` is mocked out.